app/models/user.rb

```diff
@@ -231,7 +231,29 @@ def find_confirmed_or_available_match
   def find_or_create_match
     existing_match = find_confirmed_or_available_match
     return existing_match if existing_match
-    # TODO : here look for campaigns and create match
+
+    campaigns = ::ReachableUsersService.get_running_campaigns_for_user(self)
+    return if campaigns.blank?
+    campaign = campaigns.first
+    return if campaign.blank?
+    return unless campaign.running?
+    return if campaign.remaining_doses <= 0
+    return if Time.now.utc >= campaign.ends_at
+
+    @match = Match.find_by(campaign_id: campaign.id, user_id: id)
+    if @match.blank?
+      REDIS_LOCK.lock!("create_match_for_user_id_#{id}", 2000) do
+        @match = Match.create(
```

Comment:
Does creating the match like generate an email?

Reply:
Yes